transport/src/main/java/io/netty/bootstrap/AbstractBootstrap.java

```diff
@@ -218,6 +220,12 @@ public B validate() {
         return (B) this;
     }
 
+    private void ensureChannelFactory() {
+        if (channelFactory == null) {
+            channelFactory = ChannelFactoriesRegistry.getFactoryForEventLoopGroup(group.getClass(), isServerSide());
```

Comment:
Wonder if you can get away with `(this instanceof ServerBootstrap)` instead of the `isServerSide()` method.